build_projects/shared-build-targets-utils/VersionRepoUpdater.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.DotNet.Cli.Build
+{
+    public class VersionRepoUpdater
+    {
+        private static Regex s_nugetFileRegex = new Regex("^(.*?)\\.(([0-9]+\\.)?[0-9]+\\.[0-9]+(-([A-z0-9-]+))?)\\.nupkg$");
+
+        private HttpClient _client = new HttpClient();
+        private string _gitHubUser;
+        private string _gitHubEmail;
+        private string _versionsRepoOwner;
+        private string _versionsRepo;
+
+        public VersionRepoUpdater(
+            string gitHubUser = null, 
+            string gitHubEmail = null, 
+            string versionRepoOwner = null, 
+            string versionsRepo = null, 
+            string gitHubAuthToken = null)
+        {
+            _gitHubUser = gitHubUser ?? "dotnet-bot";
+            _gitHubEmail = gitHubEmail ?? "dotnet-bot@microsoft.com";
+            _versionsRepoOwner = versionRepoOwner ?? "dotnet";
+            _versionsRepo = versionsRepo ?? "versions";
```

Comment:
`null` is definitely not valid for any of these parameters, they all need to be specified.

When I first wrote this, I was going to get these values from environment variables (like gitHubAuthToken is), which I didn't think is possible with optional parameters.

Another reasoning is how optional parameters work.  The optional value actually gets compiled into the calling assembly.  So if you were to update the optional value without recompiling all your callers, the callers still get the old optional value.  In general, to me it seems to be a good idea to make optional parameter values something that won't ever change, `null`, and then provide their defaults in actual code.